common/src/main/java/io/druid/math/expr/Parser.java

```diff
@@ -44,30 +42,40 @@
 public class Parser
 {
   private static final Logger log = new Logger(Parser.class);
-  private static final Map<String, Supplier<Function>> func;
+
+  private static final Map<String, Supplier<Function>> functions = Maps.newHashMap();
```

Comment:
Does this map need to be thread safe? I think yes, since it's written to by the initialization thread and read from query threads.